src/python/pants/task/task.py

```diff
@@ -237,8 +238,18 @@ def get_targets(self, predicate=None):
 
     :API: public
     """
-    return (self.context.targets(predicate) if self.act_transitively
-            else list(filter(predicate, self.context.target_roots)))
+    initial_targets = (self.context.targets(predicate) if self.act_transitively
+                       else list(filter(predicate, self.context.target_roots)))
+
+    included_targets = TargetFilter.scoped_instance(self).apply(initial_targets)
+    excluded_targets = set(initial_targets).difference(included_targets)
+
+    if excluded_targets:
+      self.context.log.info("{} target(s) excluded".format(len(excluded_targets)))
+      for target in excluded_targets:
+        self.context.log.debug("{} excluded".format(target.address.spec))
+
+    return included_targets
```

Comment:
I think you're already aware of this, but not many tasks use this method (they use `self.context.targets`) so there may need to be follow-up:
```
$ git grep "\bget_targets(" | cut -d: -f1 | sort -u | wc -l
17
```

Reply:
For context: This method was a relatively recent addition, to support lint/fmt tasks being optionally transitive or skipped entirely. So currently they are the only tasks that consult this method. However those tasks are also the immediate use case for this feature. Any other task we want to use it for will have to be modified to use this method, as you suggest, assuming we think that's worth doing in the v1 engine.